### PR TITLE
Update Linting Supressions and Documentation Cleanup

### PR DESCRIPTION
**Pull request by alzimmermsft**
Removes stale and over reaching linting suppressions and adds documentation and annotations to model classes.

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -202,8 +196,14 @@
...
-  <!--  -->
-  <suppress checks="LineLength" files=".*[/\\]storage[/\\].*(implementation|models)[/\\].*"/>
+  <!-- Suppress the check on any implementation code and codegen classes -->
+  <suppress checks="LineLength" files=".*[/\\]storage[/\\].*implementation[/\\].*"/>
+  <suppress checks="LineLength" files="com.azure.storage.blob.models.BlobErrorCode.java"/>
+  <suppress checks="LineLength" files="com.azure.storage.file.models.FileErrorCode.java"/>
+  <suppress checks="LineLength" files="com.azure.storage.queue.models.QueueErrorCode.java"/>
+
+  <!-- Suppress datalake for now as it is new development -->
+  <suppress checks="LineLength" files=".*[/\\]storage[/\\]file[/\\]datalake[/\\]models[/\\].*"/>
```
Comment (conniey) [inline eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: Can you create a bug for this?

Comment (alzimmermsft) [reply eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml]: I think all the models are codegen right now

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerListDetails.java [hunk 2]
```diff
@@ -11,12 +12,14 @@
  * requests, so changing the details for a different listing operation requires construction of a new object. Null may
  * be passed if none of the options are desirable.
  */
+@Fluent
 public final class BlobContainerListDetails {
-
     private boolean retrieveMetadata;
 
+    /**
+     * Constructs a unpopulated {@link BlobContainerListDetails}.
```
Comment (srnagar) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerListDetails.java]:
```suggestion
     * Constructs an unpopulated {@link BlobContainerListDetails}.
```

File: sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/models/FileUploadInfo.java [hunk 3]
```diff
@@ -3,16 +3,29 @@
...
     private final OffsetDateTime lastModified;
     private final byte[] contentMD5;
     private final Boolean isServerEncrypted;
 
+    /**
+     * Constructs a {@link FileUploadInfo}.
+     *
+     * @param eTag ETag of the file.
+     * @param lastModified Datetime when the file was last modified.
+     * @param contentMD5 MD5 of the file's content.
```
Comment (srnagar) [inline sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/models/FileUploadInfo.java]: should this be named `contentMd5`?

Comment (alzimmermsft) [reply sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/models/FileUploadInfo.java]: Done

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerListDetails.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobContainerListDetails.java]: Changed

**Comment by alzimmermsft** [comment]
Pipeline steps failed due to transient issues, merging.

**Pull request merged**